app/lib/package/api_export/export_api_to_bucket.dart

```diff
@@ -78,6 +81,83 @@ class ApiExporter {
         .write(await searchBackend.getPackageNameCompletionData());
   }
 
+  Future<void> fullSync() async {
+    final invisiblePackageNames = await dbService
+        .query<ModeratedPackage>()
+        .run()
+        .map((mp) => mp.name!)
+        .toSet();
+
+    final allPackageNames = <String>{};
+    final packageQuery = dbService.query<Package>();
+    await packageQuery.run().parallelForEach(_concurrency, (pkg) async {
+      final name = pkg.name!;
+      if (pkg.isNotVisible) {
+        invisiblePackageNames.add(name);
+        return;
+      }
+      allPackageNames.add(name);
+
+      // TODO: Consider retries around all this logic
+      await syncPackage(name);
+    });
+
+    final visibilityConflicts =
+        allPackageNames.intersection(invisiblePackageNames);
+    if (visibilityConflicts.isNotEmpty) {
```

Comment:
Probably, in which case, yes, we might just not even want to load the list of moderated packages.

And yes, that's probably better because, afaik integrity checks do take race conditions into account.